src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java

```diff
@@ -187,24 +187,47 @@ public void reportMessage(String parentGroup, String forumName, String creator,
         }
         DocumentSnapshot messageSnapshot = getForumMessage(parentGroup, forumName, creator, date);
         WriteBatch batch = documentAdapter.batch();
-        batch.update(messageSnapshot.getReference(), REPORTED_BY_FIELD, FieldValue.arrayUnion(reporter));
         ArrayList<String> messages = (ArrayList<String>) messageSnapshot.get(REPORTED_BY_FIELD);
-        if (messages != null && messages.contains(reporter)) {
-            throw new InvalidOperationException("409", "This user already reported the message");
-        }
-        if (messages != null && messages.size() > COUNTER - 1) {
-            batch.update(messageSnapshot.getReference(), BANNED_FIELD, true);
+        if (messages == null) {
+            messages = new ArrayList<>();
+            messages.add(reporter);
+            batch.update(messageSnapshot.getReference(), REPORTED_BY_FIELD, messages);
+        } else {
+            if (messages.contains(reporter)) {
+                throw new InvalidOperationException("409", "This user already reported the message");
+            }
+            batch.update(messageSnapshot.getReference(), REPORTED_BY_FIELD, FieldValue.arrayUnion(reporter));
+            if (messages.size() == COUNTER) {
+                batch.update(messageSnapshot.getReference(), BANNED_FIELD, true);
+                String usernamePath = Path.ofDocument(Collections.used_usernames, creator);
+                String uid = documentAdapter.getStringFromDocument(usernamePath, "user");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "user" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522)